### PR TITLE
Adds default process flag

### PR DESCRIPTION
**Pull request by zmackie**
Addresses: #522 

Based on the work in https://github.com/buildpacks/pack/pull/545

### CODE REVIEW

File: acceptance/acceptance_test.go [hunk 1]
```diff
@@ -1733,28 +1768,6 @@ func waitForResponse(t *testing.T, port string, timeout time.Duration) string {
...
-}
-
-func isCommandRunning(cmd *exec.Cmd) bool {
-	_, err := os.FindProcess(cmd.Process.Pid)
-	if err != nil {
-		return false
-	}
-	return true
-}
-
```
Comment (zmackie) [inline acceptance/acceptance_test.go]: Dead code removal.

File: Makefile [hunk 2]
```diff
@@ -7,6 +7,10 @@ PACKAGE_BASE=github.com/buildpacks/pack
 PACKAGES:=$(shell $(GOCMD) list ./... | grep -v /testdata/)
 SRC:=$(shell find . -type f -name '*.go' -not -path "*/vendor/*")
 ARCHIVE_NAME=pack-$(PACK_VERSION)
+TEST_TIMEOUT?=0
+UNIT_TIMEOUT?=$(TEST_TIMEOUT)
+ACCEPTANCE_TIMEOUT?=$(TEST_TIMEOUT)
+
```
Comment (zmackie) [inline Makefile]: Make test timeout adjustable.

File: build.go [hunk 3]
```diff
@@ -36,18 +36,19 @@ type Lifecycle interface {
...
+	RunImage           string              // defaults to the best mirror from the builder metadata or AdditionalMirrors
+	AdditionalMirrors  map[string][]string // only considered if RunImage is not provided
+	Env                map[string]string
+	Publish            bool
+	NoPull             bool
+	ClearCache         bool
+	Buildpacks         []string
+	ProxyConfig        *ProxyConfig // defaults to  environment proxy vars
+	ContainerConfig    ContainerConfig
+	DefaultProcessType string
```
Comment (zmackie) [inline build.go]: Make format did this

File: build.go [hunk 4]
```diff
@@ -135,17 +136,18 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
...
+		Builder:            ephemeralBuilder,
+		RunImage:           runImageName,
+		ClearCache:         opts.ClearCache,
+		Publish:            opts.Publish,
+		HTTPProxy:          proxyConfig.HTTPProxy,
+		HTTPSProxy:         proxyConfig.HTTPSProxy,
+		NoProxy:            proxyConfig.NoProxy,
+		Network:            opts.ContainerConfig.Network,
+		Volumes:            platformVolumes,
+		DefaultProcessType: opts.DefaultProcessType,
```
Comment (zmackie) [inline build.go]: `make format` did this

File: client.go [hunk 5]
```diff
@@ -103,7 +104,11 @@ func NewClient(opts ...ClientOption) (*Client, error) {
 
 	if client.docker == nil {
 		var err error
-		client.docker, err = dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithVersion("1.38"))
+		client.docker, err = dockerClient.NewClientWithOpts(
+			dockerClient.FromEnv,
+			dockerClient.WithVersion("1.38"),
+			dockerClient.WithTimeout(10*time.Second),
```
Comment (zmackie) [inline client.go]: Add timeout to docker client for robustness

File: inspect_builder.go [hunk 6]
```diff
@@ -24,8 +24,8 @@ type BuilderInfo struct {
 	CreatedBy       builder.CreatorMetadata
 }
 
-func (c *Client) InspectBuilder(name string, daemon bool) (*BuilderInfo, error) {
-	img, err := c.imageFetcher.Fetch(context.Background(), name, daemon, false)
+func (c *Client) InspectBuilder(ctx context.Context, name string, daemon bool) (*BuilderInfo, error) {
+	img, err := c.imageFetcher.Fetch(ctx, name, daemon, false)
```
Comment (zmackie) [inline inspect_builder.go]: This change (and other builder inspection changes) are basically unrelated to the feature, but I noticed they could use some refactoring was I was working. Happy to pull into their own PR.

File: internal/build/build.go [hunk 7]
```diff
@@ -51,17 +52,18 @@ func NewLifecycle(docker client.CommonAPIClient, logger logging.Logger) *Lifecyc
...
+	RunImage           string
+	ClearCache         bool
+	Publish            bool
+	HTTPProxy          string
+	HTTPSProxy         string
+	NoProxy            string
+	Network            string
+	Volumes            []string
+	DefaultProcessType string
 }
```
Comment (zmackie) [inline internal/build/build.go]: `make format` did this

**Comment by zmackie** [comment]
@jromero Seems like there are some combo failures. I haven't really wrapped my head around how that all works (like how does `pack previous` work when adding a new flag?) Any pointers there would be helpful.

**Comment by jromero** [comment]
> @jromero Seems like there are some combo failures. I haven't really wrapped my head around how that all works (like how does `pack previous` work when adding a new flag?) Any pointers there would be helpful.

Just reran tests and everything passed. I'm assuming it was an intermittent issue with GitHub.

File: client.go [hunk 8]
```diff
@@ -46,6 +47,8 @@ type Client struct {
 
 type ClientOption func(c *Client)
 
+var ClientTimeout = 1 * time.Minute
```
Comment (jromero) [inline client.go]: This shouldn't be exported. If it's intended to be configurable then it should be passed in as a `ClientOption`.

File: client.go [hunk 9]
```diff
@@ -103,7 +106,11 @@ func NewClient(opts ...ClientOption) (*Client, error) {
 
 	if client.docker == nil {
 		var err error
-		client.docker, err = dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithVersion("1.38"))
+		client.docker, err = dockerClient.NewClientWithOpts(
+			dockerClient.FromEnv,
+			dockerClient.WithVersion("1.38"),
+			dockerClient.WithTimeout(ClientTimeout),
```
[the remark this answers is not in the document]
Comment (zmackie) [reply client.go]: This was a refactor I added during the work. I'll pull it into its own PR.

File: client.go (hunk 8 above)
Comment (zmackie) [reply client.go]: Will pull into a separate PR.

**Comment by zmackie** [comment]
@jromero Now using `packSupports`

**Comment by jromero** [comment]
# Acceptance Feedback

Functionally this works really well. GJ. Minor issue with the help string. See below.

## 🤔 Improvements

### `pack build --help`

##### Expected

Help message for argument is incomplete and should start with capital letter: `set the`

##### Output

```
$ ./out/pack build --help
Generate app image from source code

Usage:
  pack build <image-name> [flags]

Flags:
  -B, --builder string           Builder image (default "cnbs/sample-builder:bionic")
  -b, --buildpack strings        Buildpack reference in the form of '<buildpack>@<version>',
                                   path to a buildpack directory (not supported on Windows),
                                   path/URL to a buildpack .tar or .tgz file, or
                                   the name of a packaged buildpack image
                                 Repeat for each buildpack in order,
                                   or supply once by comma-separated list
      --clear-cache              Clear image's associated cache before building
  -D, --default-process string   set the
  -d, --descriptor string        Path to the project descriptor file
  -e, --env stringArray          Build-time environment variable, in the form 'VAR=VALUE' or 'VAR'.
                                 When using latter value-less form, value will be taken from current
                                   environment at the time this command is executed.
                                 This flag may be specified multiple times and will override
                                   individual values defined by --env-file.
      --env-file stringArray     Build-time environment variables file
                                 One variable per line, of the form 'VAR=VALUE' or 'VAR'
                                 When using latter value-less form, value will be taken from current
                                   environment at the time this command is executed
  -h, --help                     Help for 'build'
      --network string           Connect detect and build containers to network
      --no-pull                  Skip pulling builder and run images before use
  -p, --path string              Path to app dir or zip-formatted file (defaults to current working directory)
      --publish                  Publish to registry
      --run-image string         Run image (defaults to default stack's run image)
      --volume stringArray       Mount host volume into the build container, in the form '<host path>:<target path>'. Target path will be prefixed with '/platform/'
                                 Repeat for each volume in order,
                                   or supply once by comma-separated list
```


## 👍 Good

### Default to `web` process type

```
$ ./out/pack build sample-hello-processes-app --builder cnbs/sample-builder:alpine  --buildpack ../samples/buildpacks/java-maven/ --buildpack ../samples/buildpacks/hello-processes/ --path ../samples/apps/java-maven/

alpine: Pulling from cnbs/sample-builder
Digest: sha256:f199c03ebe56d78c3a332591e9a861bd83bfbf671cd9f01308dcce74ea2093de
Status: Image is up to date for cnbs/sample-builder:alpine
alpine: Pulling from cnbs/sample-stack-run
Digest: sha256:fa05d2d97a4d7b18ddd338f093f309b6fdaff71931675544ec0fe0fdb8d90fb6
Status: Image is up to date for cnbs/sample-stack-run:alpine
Warning: buildpack io.buildpacks.samples.java-maven@0.0.1 already exists on builder and will be overwritten
===> DETECTING
[detector] io.buildpacks.samples.java-maven      0.0.1
[detector] io.buildpacks.samples.hello-processes 0.0.1
===> ANALYZING
[analyzer] Warning: Image "index.docker.io/library/sample-hello-processes-app:latest" not found
===> RESTORING
===> BUILDING
[builder] ---> Java buildpack
[builder] ---> Installing JDK
[builder] ---> Running Maven Wrapper
...
[builder] ---> Hello processes buildpack
[builder] ---> Adding sys-info process
[builder] ---> Done
===> EXPORTING
[exporter] Adding layer 'launcher'
[exporter] Adding layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Adding layer 'io.buildpacks.samples.hello-processes:sys-info'
[exporter] Adding 1/1 app layer(s)
[exporter] Adding layer 'config'
[exporter] *** Images (cdc3259f5b53):
[exporter]       index.docker.io/library/sample-hello-processes-app:latest
[exporter] Adding cache layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Adding cache layer 'io.buildpacks.samples.java-maven:maven_m2'
Successfully built image sample-hello-processes-app

$  ./out/pack inspect-image sample-hello-processes-app
Inspecting image: sample-hello-processes-app

REMOTE:
(not present)

LOCAL:

Stack: io.buildpacks.samples.stacks.alpine

Base Image:
  Reference: 44c09e95d60f583a07ee7137856511c6c88be65ba505b878ffd8c80ff6c2daaa
  Top Layer: sha256:072aabcede92b0a4f79e2906c32c0e8bcbb3a904ba1cb0f9c21f181322d286d6

Run Images:
  cnbs/sample-stack-run:alpine

Buildpacks:
  ID                                           VERSION
  io.buildpacks.samples.java-maven             0.0.1
  io.buildpacks.samples.hello-processes        0.0.1

Processes:
  TYPE                 SHELL        COMMAND                                                                   ARGS
  web (default)        bash         java -jar target/sample-0.0.1-SNAPSHOT.jar
  sys-info             bash         /layers/io.buildpacks.samples.hello-processes/sys-info/sys-info.sh


$ docker run sample-hello-processes-app
    |'-_ _-'|       ____          _  _      _                      _             _
    |   |   |      |  _ \        (_)| |    | |                    | |           (_)
     '-_|_-'       | |_) | _   _  _ | |  __| | _ __    __ _   ___ | | __ ___     _   ___
|'-_ _-'|'-_ _-'|  |  _ < | | | || || | / _` || '_ \  / _` | / __|| |/ // __|   | | / _ \
|   |   |   |   |  | |_) || |_| || || || (_| || |_) || (_| || (__ |   < \__ \ _ | || (_) |
 '-_|_-' '-_|_-'   |____/  \__,_||_||_| \__,_|| .__/  \__,_| \___||_|\_\|___/(_)|_| \___/
                                              | |
                                              |_|

:: Built with Spring Boot :: 2.1.13.RELEASE
```

### Warning when flag not supported

```
$ ./out/pack build sample-hello-processes-app --builder cnbs/sample-builder:alpine  --buildpack ../samples/buildpacks/java-maven/ --buildpack ../samples/buildpacks/hello-processes/ --path ../samples/apps/java-maven/ --default-process sys-info
alpine: Pulling from cnbs/sample-builder
Digest: sha256:f199c03ebe56d78c3a332591e9a861bd83bfbf671cd9f01308dcce74ea2093de
Status: Image is up to date for cnbs/sample-builder:alpine
alpine: Pulling from cnbs/sample-stack-run
Digest: sha256:fa05d2d97a4d7b18ddd338f093f309b6fdaff71931675544ec0fe0fdb8d90fb6
Status: Image is up to date for cnbs/sample-stack-run:alpine
Warning: buildpack io.buildpacks.samples.java-maven@0.0.1 already exists on builder and will be overwritten
===> DETECTING
[detector] io.buildpacks.samples.java-maven      0.0.1
[detector] io.buildpacks.samples.hello-processes 0.0.1
===> ANALYZING
[analyzer] Restoring metadata for "io.buildpacks.samples.java-maven:jdk" from app image
[analyzer] Restoring metadata for "io.buildpacks.samples.java-maven:maven_m2" from cache
[analyzer] Restoring metadata for "io.buildpacks.samples.hello-processes:sys-info" from app image
===> RESTORING
[restorer] Restoring data for "io.buildpacks.samples.java-maven:jdk" from cache
[restorer] Restoring data for "io.buildpacks.samples.java-maven:maven_m2" from cache
===> BUILDING
[builder] ---> Java buildpack
[builder] ---> Installing JDK
[builder] ---> Running Maven Wrapper
...
[builder] ---> Hello processes buildpack
[builder] ---> Adding sys-info process
[builder] ---> Done
===> EXPORTING
Warning: You specified a default process type but that is not supported by this version of the lifecycle
[exporter] Reusing layer 'launcher'
[exporter] Reusing layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Reusing layer 'io.buildpacks.samples.hello-processes:sys-info'
[exporter] Adding 1/1 app layer(s)
[exporter] Reusing layer 'config'
[exporter] *** Images (bbff32888cd9):
[exporter]       index.docker.io/library/sample-hello-processes-app:latest
# Buildpacks to include in builder
[exporter] Reusing cache layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Adding cache layer 'io.buildpacks.samples.java-maven:maven_m2'
Successfully built image sample-hello-processes-app

$ ./out/pack inspect-image sample-hello-processes-app
Inspecting image: sample-hello-processes-app

REMOTE:
(not present)

LOCAL:

Stack: io.buildpacks.samples.stacks.alpine

Base Image:
  Reference: 44c09e95d60f583a07ee7137856511c6c88be65ba505b878ffd8c80ff6c2daaa
  Top Layer: sha256:072aabcede92b0a4f79e2906c32c0e8bcbb3a904ba1cb0f9c21f181322d286d6

Run Images:
  cnbs/sample-stack-run:alpine

Buildpacks:
  ID                                           VERSION
  io.buildpacks.samples.java-maven             0.0.1
  io.buildpacks.samples.hello-processes        0.0.1

Processes:
  TYPE                 SHELL        COMMAND                                                                   ARGS
  web (default)        bash         java -jar target/sample-0.0.1-SNAPSHOT.jar
  sys-info             bash         /layers/io.buildpacks.samples.hello-processes/sys-info/sys-info.sh

$ docker run sample-hello-processes-app
    |'-_ _-'|       ____          _  _      _                      _             _
    |   |   |      |  _ \        (_)| |    | |                    | |           (_)
     '-_|_-'       | |_) | _   _  _ | |  __| | _ __    __ _   ___ | | __ ___     _   ___
|'-_ _-'|'-_ _-'|  |  _ < | | | || || | / _` || '_ \  / _` | / __|| |/ // __|   | | / _ \
|   |   |   |   |  | |_) || |_| || || || (_| || |_) || (_| || (__ |   < \__ \ _ | || (_) |
 '-_|_-' '-_|_-'   |____/  \__,_||_||_| \__,_|| .__/  \__,_| \___||_|\_\|___/(_)|_| \___/
                                              | |
                                              |_|

:: Built with Spring Boot :: 2.1.13.RELEASE
```

### Set default process type

```
$ ./out/pack build sample-hello-processes-app --builder custom-builder:0.7.2  --buildpack ../samples/buildpacks/java-maven/ --buildpack ../samples/buildpacks/hello-processes/ --path ../samples/apps/java-maven/ --default-process sys-info
bionic: Pulling from cnbs/sample-stack-run
Digest: sha256:f0062031b2c87fecfe5a28434dd228f950f04e03487fe15624216abe8126e8f9
Status: Image is up to date for cnbs/sample-stack-run:bionic
Warning: buildpack io.buildpacks.samples.java-maven@0.0.1 already exists on builder and will be overwritten
===> DETECTING
[detector] io.buildpacks.samples.java-maven      0.0.1
[detector] io.buildpacks.samples.hello-processes 0.0.1
===> ANALYZING
[analyzer] Restoring metadata for "io.buildpacks.samples.java-maven:jdk" from app image
[analyzer] Restoring metadata for "io.buildpacks.samples.java-maven:maven_m2" from cache
[analyzer] Restoring metadata for "io.buildpacks.samples.hello-processes:sys-info" from app image
===> RESTORING
[restorer] Restoring data for "io.buildpacks.samples.java-maven:jdk" from cache
[restorer] Restoring data for "io.buildpacks.samples.java-maven:maven_m2" from cache
===> BUILDING
[builder] ---> Java buildpack
[builder] ---> Installing JDK
[builder] ---> Running Maven Wrapper
...
[builder] ---> Hello processes buildpack
[builder] ---> Adding sys-info process
[builder] ---> Done
===> EXPORTING
[exporter] Adding layer 'launcher'
[exporter] Adding layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Adding layer 'io.buildpacks.samples.hello-processes:sys-info'
[exporter] Adding 1/1 app layer(s)
[exporter] Adding layer 'config'
[exporter] *** Images (46b12f4bf143):
[exporter]       index.docker.io/library/sample-hello-processes-app:latest
[exporter] Adding cache layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Adding cache layer 'io.buildpacks.samples.java-maven:maven_m2'
Successfully built image sample-hello-processes-app

$ ./out/pack inspect-image sample-hello-processes-app
Inspecting image: sample-hello-processes-app

REMOTE:
(not present)

LOCAL:

Stack: io.buildpacks.samples.stacks.bionic

Base Image:
  Reference: dcecb6e3b89db40c786cdbe0681b46200a22f9f2ba7aaf486df54f072cdb0c08
  Top Layer: sha256:bf3db18d60f28470d3de4408cd3c06d058f9036a791c19e2a1831ae2a88770a4

Run Images:
  cnbs/sample-stack-run:bionic

Buildpacks:
  ID                                           VERSION
  io.buildpacks.samples.java-maven             0.0.1
  io.buildpacks.samples.hello-processes        0.0.1

Processes:
  TYPE                      SHELL        COMMAND                                                                   ARGS
  sys-info (default)        bash         /layers/io.buildpacks.samples.hello-processes/sys-info/sys-info.sh
  web                       bash         java -jar target/sample-0.0.1-SNAPSHOT.jar

$ docker run sample-hello-processes-app
     env vars:
       declare -x CNB_GROUP_ID="1000"
       declare -x CNB_USER_ID="1000"
       declare -x HOME="/home/cnb"
       declare -x HOSTNAME="ff6027eeceae"
       declare -x JAVA_HOME="/layers/io.buildpacks.samples.java-maven/jdk"
       declare -x LD_LIBRARY_PATH="/layers/io.buildpacks.samples.java-maven/jdk/jre/lib/amd64/server:"
       declare -x OLDPWD
       declare -x PATH="/layers/io.buildpacks.samples.java-maven/jdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
       declare -x PWD="/workspace"
       declare -x SHLVL="2"
```

### Attempting to set a wrong process type

```
$ ./out/pack build sample-hello-processes-app --builder custom-builder:0.7.2  --buildpack ../samples/buildpacks/java-maven/ --buildpack ../samples/buildpacks/hello-processes/ --path ../samples/apps/java-maven/ --default-process unknown-process-type
bionic: Pulling from cnbs/sample-stack-run
Digest: sha256:f0062031b2c87fecfe5a28434dd228f950f04e03487fe15624216abe8126e8f9
Status: Image is up to date for cnbs/sample-stack-run:bionic
Warning: buildpack io.buildpacks.samples.java-maven@0.0.1 already exists on builder and will be overwritten
===> DETECTING
[detector] io.buildpacks.samples.java-maven      0.0.1
[detector] io.buildpacks.samples.hello-processes 0.0.1
===> ANALYZING
[analyzer] Restoring metadata for "io.buildpacks.samples.java-maven:jdk" from app image
[analyzer] Restoring metadata for "io.buildpacks.samples.java-maven:maven_m2" from cache
[analyzer] Restoring metadata for "io.buildpacks.samples.hello-processes:sys-info" from app image
===> RESTORING
[restorer] Restoring data for "io.buildpacks.samples.java-maven:jdk" from cache
[restorer] Restoring data for "io.buildpacks.samples.java-maven:maven_m2" from cache
===> BUILDING
[builder] ---> Java buildpack
[builder] ---> Installing JDK
[builder] ---> Running Maven Wrapper
...
[builder] ---> Hello processes buildpack
[builder] ---> Adding sys-info process
[builder] ---> Done
===> EXPORTING
[exporter] Reusing layer 'launcher'
[exporter] Reusing layer 'io.buildpacks.samples.java-maven:jdk'
[exporter] Reusing layer 'io.buildpacks.samples.hello-processes:sys-info'
[exporter] Adding 1/1 app layer(s)
[exporter] Reusing layer 'config'
[exporter] ERROR: failed to export: default process type 'unknown-process-type' not present in list [sys-info web]
ERROR: failed with status code: 1
```

File: internal/commands/build.go [hunk 10]
```diff
@@ -119,6 +121,7 @@ func buildCommandFlags(cmd *cobra.Command, buildFlags *BuildFlags, cfg config.Co
 	cmd.Flags().StringVar(&buildFlags.Network, "network", "", "Connect detect and build containers to network")
 	cmd.Flags().StringVarP(&buildFlags.DescriptorPath, "descriptor", "d", "", "Path to the project descriptor file")
 	cmd.Flags().StringArrayVar(&buildFlags.Volumes, "volume", nil, "Mount host volume into the build container, in the form '<host path>:<target path>'. Target path will be prefixed with '/platform/'"+multiValueHelp("volume"))
+	cmd.Flags().StringVarP(&buildFlags.DefaultProcessType, "default-process", "D", "", "set the ")
```
Comment (jromero) [inline internal/commands/build.go]:
```suggestion
	cmd.Flags().StringVarP(&buildFlags.DefaultProcessType, "default-process", "D", "", "Set the default process type")
```

Comment (jromero) [reply internal/commands/build.go]: @zmackie Seems like the sign-off was not included. :/

Comment (zmackie) [reply internal/commands/build.go]: fixed it.

**Comment by zmackie** [comment]
@jromero I think we're GTG on this one. Wanna take a final 👀 ?

**Comment by jromero** [comment]
# ✅ Accepted

### Help

```
./out/pack build --help
Generate app image from source code

Usage:
  pack build <image-name> [flags]

Flags:
  -B, --builder string           Builder image (default "cnbs/sample-builder:bionic")
  -b, --buildpack strings        Buildpack reference in the form of '<buildpack>@<version>',
                                   path to a buildpack directory (not supported on Windows),
                                   path/URL to a buildpack .tar or .tgz file, or
                                   the name of a packaged buildpack image
                                 Repeat for each buildpack in order,
                                   or supply once by comma-separated list
      --clear-cache              Clear image's associated cache before building
  -D, --default-process string   Set the default process type
  -d, --descriptor string        Path to the project descriptor file
  -e, --env stringArray          Build-time environment variable, in the form 'VAR=VALUE' or 'VAR'.
                                 When using latter value-less form, value will be taken from current
                                   environment at the time this command is executed.
                                 This flag may be specified multiple times and will override
                                   individual values defined by --env-file.
      --env-file stringArray     Build-time environment variables file
                                 One variable per line, of the form 'VAR=VALUE' or 'VAR'
                                 When using latter value-less form, value will be taken from current
                                   environment at the time this command is executed
  -h, --help                     Help for 'build'
      --network string           Connect detect and build containers to network
      --no-pull                  Skip pulling builder and run images before use
  -p, --path string              Path to app dir or zip-formatted file (defaults to current working directory)
      --publish                  Publish to registry
      --run-image string         Run image (defaults to default stack's run image)
      --volume stringArray       Mount host volume into the build container, in the form '<host path>:<target path>'. Target path will be prefixed with '/platform/'
                                 Repeat for each volume in order,
                                   or supply once by comma-separated list

Global Flags:
      --no-color     Disable color output
  -q, --quiet        Show less output
      --timestamps   Enable timestamps in output
  -v, --verbose      Show more output

```

**Pull request merged**